zkserver/zkserver.go

```diff
@@ -116,6 +127,36 @@ func (z *ZKS) writeMessage(kx *session.KX, msg *RPCWrapper) error {
 	return nil
 }
 
+func (z *ZKS) unwelcome(kx *session.KX, reason string) error {
+	// assemble command
+	message := rpc.Message{
+		Command: rpc.SessionCmdUnwelcome,
+	}
+	payload := rpc.Unwelcome{
+		Version: rpc.ProtocolVersion,
+		Reason:  reason,
+	}
+
+	// encode command
+	var bb bytes.Buffer
+	_, err := xdr.Marshal(&bb, message)
+	if err != nil {
+		return fmt.Errorf("could not marshal Unwelcome message")
+	}
+	_, err = xdr.Marshal(&bb, payload)
+	if err != nil {
+		return fmt.Errorf("could not marshal Unwelcome payload")
```

Comment:
: %v", err)

Reply:
XDR errors spit the input out. It is very loud and provides no utility to print it. For example, a chunk is 256KB and if we print it it'll be more than 1MB worth of text. No one wants to see that.